File: 백준/Silver/2606. 바이러스/바이러스.py
def dfs(v):
    visited[v] = 1
    for nx in graph[v]:
        if visited[nx] == 0:
            dfs(nx)

n = int(input())
m = int(input())

graph = [[] for _ in range(n + 1)]
visited = [0] * (n + 1)

for _ in range(m):
    a, b = map(int, input().split())
    graph[a] += [b]
    graph[b] += [a]

dfs(1)
print(sum(visited) - 1)

File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#include <vector>
using namespace std;

vector<int> graph[101];
bool visited[101];
int cnt = 0;

void dfs(int i){
    visited[i] = true;
    for(int j = 0; j < graph[i].size(); j++){
        int k = graph[i][j];
        if(!visited[k]){
            dfs(k);
            cnt++;
        }
    }
}

int main(void){
    int n, m;
    cin >> n >> m;

    while(m--){
        int a, b;
        cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }
    
    dfs(1);

    cout << cnt << endl;

    return 0;
}